src/Processors/Formats/Impl/PrettyBlockOutputFormat.h

```diff
@@ -50,6 +53,38 @@ class PrettyBlockOutputFormat : public IOutputFormat
     void writeValueWithPadding(
         const IColumn & column, const ISerialization & serialization, size_t row_num,
         size_t value_width, size_t pad_to_width, bool align_right);
+
+private:
+    bool mono_block;
+    /// For mono_block == true only
+    Chunk mono_chunk;
 };
 
+template <class OutputFormat>
+void registerPrettyFormatWithNoEscapesAndMonoBlock(FormatFactory & factory, const String & base_name)
+{
+    for (bool no_escapes : {true, false})
+    {
+        String name = no_escapes ? base_name + "NoEscapes" : base_name;
+        for (bool mono_block : {true, false})
+        {
+            factory.registerOutputFormat(mono_block ? name + "MonoBlock" : name, [no_escapes, mono_block](
+                WriteBuffer & buf,
+                const Block & sample,
+                const RowOutputFormatParams &,
+                const FormatSettings & format_settings)
+            {
+                if (no_escapes)
+                {
+                    FormatSettings changed_settings = format_settings;
+                    changed_settings.pretty.color = false;
+                    return std::make_shared<OutputFormat>(buf, sample, changed_settings, mono_block);
+                }
+                return std::make_shared<OutputFormat>(buf, sample, format_settings, mono_block);
+            });
+        }
+        factory.markOutputFormatSupportsParallelFormatting(name);
```

Comment:
Is it outside of inner loop because mono_block shouldn't enable `ParallelFormatting`?